### ISSUE TITLE
Scoold won't attempt to reconnect to Para if connection failed upon startup

### ISSUE THREAD

**Issue by bviktor** [Erudika/scoold#106]
Title says it all. I realized this after writing the basic systemd units for CentOS. It keeps saying unable to connect to Para service, even though Para's already running, it just didn't finish starting up the moment Para was starting up.

---

If, however, Para is stopped _after_ they both successfully started, Scoold will display nothing but a white page, and this appears in stdout:

```
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: 2019-02-18 20:13:01 [WARN ] /
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException: MessageBodyReader not found for media type=text/html, type=interface java.util.Map, genericType=interface java.util.Map.
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1013)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:897)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at javax.servlet.http.HttpServlet.service(HttpServlet.java:645)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:882)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:867)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1623)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at com.erudika.scoold.utils.CsrfFilter.doFilter(CsrfFilter.java:88)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:99)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:92)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:93)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:200)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1610)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1588)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1557)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.Server.handle(Server.java:502)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at java.lang.Thread.run(Thread.java:748)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: Caused by: org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException: MessageBodyReader not found for media type=text/html, type=interface java.util.Map, genericType=interface java.util.Map.
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:232)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:156)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1091)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:874)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:808)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:321)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.InboundJaxrsResponse$1.call(InboundJaxrsResponse.java:115)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.internal.Errors.process(Errors.java:229)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:389)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.InboundJaxrsResponse.runInScopeIfPossible(InboundJaxrsResponse.java:264)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.InboundJaxrsResponse.readEntity(InboundJaxrsResponse.java:112)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at com.erudika.para.client.ParaClient.getEntity(ParaClient.java:283)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at com.erudika.para.client.ParaClient.me(ParaClient.java:1296)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at com.erudika.scoold.utils.ScooldUtils.checkAuth(ScooldUtils.java:123)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at com.erudika.scoold.utils.ScooldRequestInterceptor.preHandle(ScooldRequestInterceptor.java:59)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.HandlerExecutionChain.applyPreHandle(HandlerExecutionChain.java:136)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1033)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:942)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1005)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: ... 49 common frames omitted
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: 2019-02-18 20:13:01 [WARN ] EXCEPTION
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException: MessageBodyReader not found for media type=text/html, type=interface java.util.Map, genericType=interface java.util.Map.
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1013)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:897)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at javax.servlet.http.HttpServlet.service(HttpServlet.java:645)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:882)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:867)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:542)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:566)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1588)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1557)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:227)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.Dispatcher.error(Dispatcher.java:86)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ErrorHandler.doError(ErrorHandler.java:119)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ErrorHandler.handle(ErrorHandler.java:78)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.springframework.boot.web.embedded.jetty.JettyEmbeddedErrorHandler.handle(JettyEmbeddedErrorHandler.java:55)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.Response.sendError(Response.java:655)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.AbstractHandler.doError(AbstractHandler.java:100)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1339)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1557)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
Feb 18 20:13:01 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.Server.handle(Server.java:502)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:419)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at java.lang.Thread.run(Thread.java:748)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: Caused by: org.glassfish.jersey.message.internal.MessageBodyProviderNotFoundException: MessageBodyReader not found for media type=text/html, type=interface java.util.Map, genericType=interface java.util.Map.
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:232)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:156)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1091)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:874)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:808)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:321)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.InboundJaxrsResponse$1.call(InboundJaxrsResponse.java:115)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.internal.Errors.process(Errors.java:316)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.internal.Errors.process(Errors.java:298)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.internal.Errors.process(Errors.java:229)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:389)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.InboundJaxrsResponse.runInScopeIfPossible(InboundJaxrsResponse.java:264)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.glassfish.jersey.client.InboundJaxrsResponse.readEntity(InboundJaxrsResponse.java:112)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at com.erudika.para.client.ParaClient.getEntity(ParaClient.java:283)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at com.erudika.para.client.ParaClient.me(ParaClient.java:1296)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at com.erudika.scoold.utils.ScooldUtils.checkAuth(ScooldUtils.java:123)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at com.erudika.scoold.utils.ScooldRequestInterceptor.preHandle(ScooldRequestInterceptor.java:59)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.HandlerExecutionChain.applyPreHandle(HandlerExecutionChain.java:136)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1033)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:942)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1005)
Feb 18 20:13:02 hth.ad.fooworks.com java[5450]: ... 48 common frames omitted
```

The moment I restart Para this error is gone and Scoold runs as before, without having to restart Scoold itself.

So in essence, it displays the cannot connect to para message when it shouldn't, and won't display it when it should.

Can you confirm this or am I missing something?

**Comment by albogdano** [comment]
Yes, this is the current behavior of Scoold. The error message on disconnect could be improved, I agree. I can also make it retry the connection on startup N times in an interval of 10s. That should solve the problem.

**Comment by bviktor** [comment]
Sounds good! I also added timeouts to the systemd units to make sure they start up properly during startup, but it won't hurt if Scoold also tries to heal itself I guess.

**Comment by albogdano** [comment]
Implemented in 79f3f46

**Comment by bviktor** [comment]
Actually, the reconnect works now, but now Scoold won't display the "failed to connect to Para" message at all when Para's stopped. When you open Scoold, you're signed out on the site, and when you try to sign in it'll fail because it tries to redirect you to the backend, which is not running (another reason why I believe the client should never access the backend directly).
